cmd/operator/run.go

```diff
@@ -68,44 +59,36 @@ func runProtoform(configPath string) {
 	}
 
 	// Log Kubernetes version
-	kversion, err := bdutil.GetKubernetesVersion(deployer.KubeClientSet)
+	kversion, err := util.GetKubernetesVersion(deployer.KubeClientSet)
 	if err == nil {
-		logrus.Infof("Kubernetes: %s", kversion)
+		log.Infof("Kubernetes: %s", kversion)
 	}
 
 	// Log Openshift version
-	oversion, err := bdutil.GetOcVersion(deployer.KubeClientSet)
+	oversion, err := util.GetOcVersion(deployer.KubeClientSet)
 	if err == nil {
-		logrus.Infof("Openshift: %s", oversion)
+		log.Infof("Openshift: %s", oversion)
 	}
 
 	stopCh := make(chan struct{})
 
-	//sampleController := sample.NewCRDInstaller(deployer.Config, deployer.KubeConfig, deployer.KubeClientSet, bdutil.GetSampleDefaultValue(), stopCh)
-	//deployer.AddController(sampleController)
-
-	alertController := alert.NewCRDInstaller(deployer.Config, deployer.KubeConfig, deployer.KubeClientSet, bdutil.GetAlertTemplate(), stopCh)
-	deployer.AddController(alertController)
-
-	hubController := blackduck.NewCRDInstaller(deployer.Config, deployer.KubeConfig, deployer.KubeClientSet, bdutil.GetBlackDuckTemplate(), stopCh)
-	deployer.AddController(hubController)
-
-	opssSightController, err := opssight.NewCRDInstaller(&opssight.Config{
-		Config:        deployer.Config,
-		KubeConfig:    deployer.KubeConfig,
-		KubeClientSet: deployer.KubeClientSet,
-		Defaults:      bdutil.GetOpsSightDefault(),
-		Threadiness:   deployer.Config.Threadiness,
-		StopCh:        stopCh,
-	})
-	if err != nil {
-		panic(err)
+	var isClusterScoped bool
+	clusterScopeEnv, ok := os.LookupEnv("CLUSTER_SCOPE")
+	if ok && len(clusterScopeEnv) > 0 {
+		clusterScope, err := strconv.ParseBool(clusterScopeEnv)
+		if err != nil {
+			panic("cluster scope environment variable is not set properly. possible values are true/false")
+		}
+		isClusterScoped = clusterScope
 	}
-	deployer.AddController(opssSightController)
-
-	logrus.Info("Starting deployer.  All controllers have been added to Protoform.")
-	if err = deployer.Deploy(); err != nil {
-		logrus.Errorf("ran into errors during deployment, but continuing anyway: %s", err.Error())
+	crdEnv, ok := os.LookupEnv("CRD_NAMES")
+	if ok && len(crdEnv) > 0 {
+		crds := strings.Split(crdEnv, ",")
```

Comment:
Maybe could use `crdEnv = strings.Trim(crdEnv, ",")` just for tricky usages with leading and/or trailing commas, but nevermind I see there is case statement that would catch it further down

Reply:
done